tools/nni_gpu_tool/gpu_metrics_collector.py

```diff
@@ -10,27 +10,32 @@
 
 from xml.dom import minidom
 
+
 def check_ready_to_run():
     if sys.platform == 'win32':
         pgrep_output = subprocess.check_output(
             'wmic process where "CommandLine like \'%nni_gpu_tool.gpu_metrics_collector%\' and name like \'%python%\'" get processId')
         pidList = pgrep_output.decode("utf-8").strip().split()
-        pidList.pop(0) # remove the key word 'ProcessId'
+        pidList.pop(0)  # remove the key word 'ProcessId'
         pidList = list(map(int, pidList))
         pidList.remove(os.getpid())
         return not pidList
     else:
-        pgrep_output = subprocess.check_output('pgrep -fxu "$(whoami)" \'python3 -m nni_gpu_tool.gpu_metrics_collector\'', shell=True)
+        pgrep_output = subprocess.check_output(
+            'pgrep -afu "$(whoami)" \'python3 -m nni_gpu_tool.gpu_metrics_collector\'', shell=True)
```

Comment:
why use `-a` instead of `-x`? 
`

Reply:
-x means exact match and -a means output command text also. It works with default Python, but if user uses multiple Python tools, like pyenv, the command line won't be exact match like this. So use a more complex filter to handle this situation.